### PR TITLE
Remove unnecessary lodash uses

### PR DESCRIPTION
**Pull request by jimmycallin**
### Reasons for making this change

Just removing some lodash uses that is more efficiently served by built-in functions, to reduce bundle size.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/antd/src/templates/ObjectFieldTemplate/index.js [hunk 1]
```diff
@@ -62,12 +61,12 @@ const ObjectFieldTemplate = ({
         ? 24
         : 12;
 
-    if (_.isObject(colSpan)) {
+    if (colSpan instanceof Object) {
```
Comment (heath-freenome) [inline packages/antd/src/templates/ObjectFieldTemplate/index.js:64]: I'm pretty sure this isn't a 1:1 change given that `isObject` is implemented as:
```
function isObject(value) {
  var type = typeof value;
  return value != null && (type == 'object' || type == 'function');
}
```

File: packages/validator-ajv6/src/createAjvInstance.ts [hunk 2]
```diff
@@ -42,7 +41,7 @@ export default function createAjvInstance(
   }
 
   // add more custom formats to validate against
-  if (isObject(customFormats)) {
+  if (customFormats instanceof Object) {
```
Comment (heath-freenome) [inline packages/validator-ajv6/src/createAjvInstance.ts:44]: I'm pretty sure this isn't a 1:1 change given that `isObject` is implemented as:
```
function isObject(value) {
  var type = typeof value;
  return value != null && (type == 'object' || type == 'function');
}
```

File: packages/antd/src/templates/ObjectFieldTemplate/index.js [hunk 3]
```diff
@@ -62,12 +61,12 @@ const ObjectFieldTemplate = ({
...
       return (
         colSpan[widget] || colSpan[field] || colSpan[type] || defaultColSpan
       );
     }
-    if (_.isNumber(colSpan)) {
+    if (typeof colSpan === "number") {
```
Comment (heath-freenome) [inline packages/antd/src/templates/ObjectFieldTemplate/index.js:69]: I'm pretty sure this isn't a 1:1 change given that `isNumber` is implemented as:
```
function isNumber(value) {
  return typeof value == 'number' ||
    (isObjectLike(value) && baseGetTag(value) == numberTag);
}
```

File: packages/utils/src/mergeSchemas.ts [hunk 4]
```diff
@@ -32,7 +30,7 @@ export default function mergeSchemas(
       Array.isArray(right)
     ) {
       // Don't include duplicate values when merging 'required' fields.
-      acc[key] = union(left, right);
```
Comment (heath-freenome) [inline packages/utils/src/mergeSchemas.ts:35]: I'm pretty sure this isn't a 1:1 as the `lodash` `union()` handles more use cases

**Comment by heath-freenome** [comment]
@jimmycallin How did you want to deal with the differences I pointed out above?

**Comment by jimmycallin** [comment]
@heath-freenome thanks for the comments, I reverted those changes to reduce risk of missing use cases.

File: packages/utils/src/mergeSchemas.ts [hunk 5]
```diff
@@ -1,5 +1,4 @@
 import union from "lodash/union";
-
```
Comment (heath-freenome) [inline packages/utils/src/mergeSchemas.ts:2]: Revert this one line change so that then entire file is no longer in the PR

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**